### PR TITLE
conn_pool: renames (request -> stream)

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low (renames)
Testing: tests renamed too
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: source/common/conn_pool/conn_pool_base.cc [hunk 1]
```diff
@@ -161,16 +161,16 @@ ConnectionPool::Cancellable* ConnPoolImplBase::newStream(AttachContext& context)
...
     // Even if there's a ready client, we may want to prefetch a new connection
     // to handle the next incoming stream.
     tryCreateNewConnections();
     return nullptr;
   }
 
   if (host_->cluster().resourceManager(priority_).pendingRequests().canCreate()) {
-    ConnectionPool::Cancellable* pending = newPendingRequest(context);
-    // This must come after newPendingRequest() because this function uses the
+    ConnectionPool::Cancellable* pending = newPendingStream(context);
```
Comment (asraa) [inline source/common/conn_pool/conn_pool_base.cc:172]: I can't comment on the line above this, but will the resource manager's requests also be renamed?

**Review by asraa — COMMENTED** [review]
Thanks! Is this re-name only targetting the connection pool component (and what about OriginalConnPool?) Otherwise LGTM

**Comment by alyssawilk** [comment]
Yeah, OriginalConnPool is theoretically going away so I had no intent to clean that up.
Unfortunately there's a bunch of lingering request terminology scattered through the code base, some of which is in stats or APIs so is going to stick around. I think I just signed up for  [new] conn pool cleanup :-)

**Review by asraa — APPROVED** [review]
Thanks for the clarification! LGTM

**Pull request merged**